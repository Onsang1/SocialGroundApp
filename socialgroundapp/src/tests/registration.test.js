/**
* @jest-environment jsdom
*/

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import Registration from '../components/registration';

test('renders Sign Up button', () => {
  render(
    <Router>
      <Registration />
    </Router>,
  );
  const linkElement = screen.getByRole('button', { name: /Sign Up/i });
  expect(linkElement).toBeInTheDocument();
});

test('renders Log In link', () => {
  render(
    <Router>
      <Registration />
    </Router>,
  );
  const linkElement = screen.getByText(/Log/);
  expect(linkElement).toBeInTheDocument();
});

test('registration page matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <Registration />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('page to have logo image', () => {
  render(
    <Router>
      <Registration />
    </Router>,
  );
  expect(screen.getByRole('img')).toBeInTheDocument();
});

test('Textbox to have value after input', async () => {
  render(
    <Router>
      <Registration />
    </Router>,
  );

  const emailLabel = screen.getByText(/Email/);
  const passwordLabel = screen.getAllByText(/Password/)[0];

  expect(emailLabel).toBeInTheDocument();
  expect(passwordLabel).toBeInTheDocument();

  const emailInput = screen.getByLabelText(/Email/);
  expect(emailInput).toHaveAttribute('type', 'text');

  const element = screen.getByRole('textbox', { name: /Email/ });

  await userEvent.type(element, 'username123@gmail.com');
  expect(screen.getByLabelText(/Email/)).toHaveValue('username123@gmail.com');
});

test('Error message should show for blank fields, invalid email format, or password', async () => {
  render(
    <Router>
      <Registration />
    </Router>,
  );

  const nameElement = screen.findAllByText(/Full/)[0];
  const emailElement = screen.getByRole('textbox', { name: /Email/ });
  const pwElement = screen.getAllByText(/Password/)[0];
  const confirmPwElement = screen.findAllByText(/Confirm/)[0];
  expect(screen.getByRole('button')).toBeInTheDocument();

  await userEvent.type(nameElement, 'user111');
  await userEvent.type(emailElement, 'user111@gmail.com');
  await userEvent.click(screen.getByRole('button'));
  expect(screen.getByText(/missing/)).toBeInTheDocument();

  await userEvent.type(nameElement, 'user111');
  await userEvent.type(emailElement, 'username111@');
  await userEvent.type(pwElement, 'password111');
  await userEvent.type(confirmPwElement, 'password111');
  await userEvent.click(screen.getByRole('button'));
  expect(screen.getByText(/format/)).toBeInTheDocument();

  await userEvent.type(nameElement, 'user111');
  await userEvent.type(emailElement, 'username111@gmail.com');
  await userEvent.type(pwElement, 'password111');
  await userEvent.type(confirmPwElement, 'password');
  await userEvent.click(screen.getByRole('button'));
  expect(screen.getByText(/match/)).toBeInTheDocument();
});
