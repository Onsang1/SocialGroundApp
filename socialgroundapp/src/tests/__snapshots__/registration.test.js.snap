// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration page matches snapshot 1`] = `
<div
  id="react-root"
>
  <div
    className="login"
  >
    <div
      className="form-container"
    >
      <form
        className="login-form"
        id="login-form"
        method="post"
        onSubmit={[Function]}
      >
        <div
          className="large-sg-header-homepage-redirect"
        >
          <a
            href="/"
          >
            <div
              aria-disabled="false"
            >
              <img
                alt="large_logo"
                className="large-logo"
                src={Object {}}
              />
            </div>
          </a>
        </div>
        <h2
          className="registrationMessage"
        >
          Sign up to see photos and videos from others.
        </h2>
        <div
          className="regMsg"
          id="regMsg"
        >
          <div
            className="regMsg"
            data-testid="blankregMsg"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <h4>
              One or more fields is/are missing. Please fill in all fields.
            </h4>
          </div>
          <div
            className="regMsg"
            data-testid="emailErrorRegMsg"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <h4>
              Your email is in an invalid format. Please re-enter.
            </h4>
          </div>
          <div
            className="regMsg"
            data-testid="pwRegMsg"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <h4>
              Your password entries do not match. Please re-enter.
            </h4>
          </div>
        </div>
        <div
          className="form-divider"
        >
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput6"
              >
                <span
                  className="form-fields"
                >
                  Full Name
                </span>
                <input
                  aria-label="Full Name"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regNameUpdate"
                  id="formInput6"
                  maxLength="100"
                  name="newName"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput12"
              >
                <span
                  className="form-fields"
                >
                  Username
                </span>
                <input
                  aria-label="Username"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regUsernameUpdate"
                  id="formInput12"
                  maxLength="100"
                  name="newUsername"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput7"
              >
                <span
                  className="form-fields"
                >
                  Email
                </span>
                <input
                  aria-label="Email"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regEmailUpdate"
                  id="formInput7"
                  maxLength="100"
                  name="newEmail"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput8"
              >
                <span
                  className="form-fields"
                >
                  Password
                </span>
                <input
                  aria-label="Password"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regPwUpdate"
                  id="formInput8"
                  maxLength="100"
                  name="newPassword"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput9"
              >
                <span
                  className="form-fields"
                >
                  Confirm Password
                </span>
                <input
                  aria-label="Confirm Password"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regConfirmPwUpdate"
                  id="formInput9"
                  maxLength="100"
                  name="confirmPassword"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </label>
            </div>
          </div>
          <br />
          <div
            className="form-submit"
          >
            <button
              className="submit-button"
              type="submit"
            >
              <div
                className="button-label"
              >
                Sign Up
              </div>
            </button>
          </div>
        </div>
        <div
          className="log-in-container"
        >
          <p
            className="log-in-redirect"
          >
            Already have an account?Â 
            <a
              href="/"
            >
              <span
                className="log-in"
                id="regSign"
              >
                Log In
              </span>
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`test to see if registration components are rendered 1`] = `
<div
  id="react-root"
>
  <div
    className="login"
  >
    <div
      className="form-container"
    >
      <form
        className="login-form"
        id="login-form"
        method="post"
        onSubmit={[Function]}
      >
        <div
          className="large-sg-header-homepage-redirect"
        >
          <a
            href="/"
          >
            <div
              aria-disabled="false"
            >
              <img
                alt="large_logo"
                className="large-logo"
                src={Object {}}
              />
            </div>
          </a>
        </div>
        <h2
          className="registrationMessage"
        >
          Sign up to see photos and videos from others.
        </h2>
        <div
          className="regMsg"
          id="regMsg"
        >
          <div
            className="regMsg"
            data-testid="blankregMsg"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <h4>
              One or more fields is/are missing. Please fill in all fields.
            </h4>
          </div>
          <div
            className="regMsg"
            data-testid="emailErrorRegMsg"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <h4>
              Your email is in an invalid format. Please re-enter.
            </h4>
          </div>
          <div
            className="regMsg"
            data-testid="pwRegMsg"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <h4>
              Your password entries do not match. Please re-enter.
            </h4>
          </div>
        </div>
        <div
          className="form-divider"
        >
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput6"
              >
                <span
                  className="form-fields"
                >
                  Full Name
                </span>
                <input
                  aria-label="Full Name"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regNameUpdate"
                  id="formInput6"
                  maxLength="100"
                  name="newName"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput12"
              >
                <span
                  className="form-fields"
                >
                  Username
                </span>
                <input
                  aria-label="Username"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regUsernameUpdate"
                  id="formInput12"
                  maxLength="100"
                  name="newUsername"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput7"
              >
                <span
                  className="form-fields"
                >
                  Email
                </span>
                <input
                  aria-label="Email"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regEmailUpdate"
                  id="formInput7"
                  maxLength="100"
                  name="newEmail"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput8"
              >
                <span
                  className="form-fields"
                >
                  Password
                </span>
                <input
                  aria-label="Password"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regPwUpdate"
                  id="formInput8"
                  maxLength="100"
                  name="newPassword"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="form-row-container"
          >
            <div
              className="form-cell"
            >
              <label
                className="form-entry-box"
                htmlFor="formInput9"
              >
                <span
                  className="form-fields"
                >
                  Confirm Password
                </span>
                <input
                  aria-label="Confirm Password"
                  aria-required="true"
                  autoCapitalize="off"
                  autoCorrect="off"
                  className="form-field-entry"
                  data-testid="regConfirmPwUpdate"
                  id="formInput9"
                  maxLength="100"
                  name="confirmPassword"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </label>
            </div>
          </div>
          <br />
          <div
            className="form-submit"
          >
            <button
              className="submit-button"
              type="submit"
            >
              <div
                className="button-label"
              >
                Sign Up
              </div>
            </button>
          </div>
        </div>
        <div
          className="log-in-container"
        >
          <p
            className="log-in-redirect"
          >
            Already have an account?Â 
            <a
              href="/"
            >
              <span
                className="log-in"
                id="regSign"
              >
                Log In
              </span>
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</div>
`;
